ocaml/CMakeLists.txt

```diff
@@ -59,11 +60,67 @@ if (NOT OCAMLC_FOUND)
     DEPENDS ${OCAMLC}
     COMMENT "Compiling ocamlbuild")
 
+  add_custom_command(
+    OUTPUT ${OPAM}
+    COMMAND /bin/bash -c 'PATH="${OCAML_PREFIX}/bin:$ENV{PATH}" ./configure --prefix="${OCAML_PREFIX}"'
+    # BEGIN PATCHING extlib
+    # This allows us to skip directly modifying opam's Makefile or opam's src_ext Makefile
+    COMMAND ${CMAKE_COMMAND} -E ARGS remove_directory src_ext/extlib
```

Comment:
Sorry, I meant can most of these subprocesses be avoided?

e.g. use add_custom_command and things like FILE(REMOVE_RECURSIVE instead of shelling out; this is still basically a more verbose alternative to writing a standalone shell script.